### ISSUE TITLE
Tag class does not implement assoc protocol

### ISSUE THREAD

**Issue by hyphz** [slavapestov/factor#1288]
The documentation for the tag class states that it "implements the assoc protocol by acting like [the tag's] attributes", but it does not do so:

Create a test tag

     IN: scratchpad "test" { { "name" "bob" } } { } <tag>
     --- Data stack:
     T{ tag f ~name~ ~attrs~ ~array~ }

Confirm the tag was created and has attribute name:

     IN: scratchpad dup pprint-xml
     <test name="bob"> 
     </test>
     --- Data stack:
     T{ tag f ~name~ ~attrs~ ~array~ }

Now trying to access the attribute via the assoc protocol fails:

     IN: scratchpad "name" swap at*  
     --- Data stack:
     f
     f

**Comment by mrjbq7** [comment]
This fixes it:

```factor
! They also follow the assoc protocol (for attributes)
CONSULT: assoc-protocol tag attrs>> ;
INSTANCE: tag assoc
```

@erg should we add this or remove the docs?

**Comment by mrjbq7** [comment]
By the way, thanks @hyphz for reporting these issues, appreciate the feedback!  I didn't realize until I looked at the code that ``tag`` implements ``sequence`` protocol for access to children and (if this patch is agreeable) the ``assoc`` protocol for access to attributes.  Kinda neat.

**Comment by mrjbq7** [comment]
Fixed by 102aa2cbc12cc5f50a82ae922f832bd02e1d017f.
